tools/find_deprecated.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""List deprecated decorators."""
+
+from pathlib import Path
+from collections import OrderedDict, defaultdict
+import ast
+from datetime import datetime
+import requests
+
+
+class Deprecation:
```

Comment:
Consider adding type hints to this file. While it's indeed only a tool, I encourage treating build-support code as first-class.

Even though we're not using MyPy, type hints will still help with making this code more self-documenting. And it will save future us time if/when we add MyPy.

Reply:
with the help of @mrossinek , type hints in 56904d516426426af96c81286158911eadf3444a !